dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeFactory.java

```diff
@@ -246,21 +250,52 @@ public static void saveMultiTree(MultiTree mTree) throws DotDataException {
      * @throws DotSecurityException
      */
     @WrapInTransaction
-    public static void saveMultiTrees(List<MultiTree> mTrees) throws DotDataException {
+    public static void saveMultiTrees(final List<MultiTree> mTrees) throws DotDataException {
         if (mTrees == null || mTrees.isEmpty())
             throw new DotDataException("empty list passed in");
         int i = 0;
         for (MultiTree tree : mTrees) {
             _dbUpsert(tree.setTreeOrder(i++));
         }
 
-        MultiTree mTree = mTrees.get(0);
+        final MultiTree mTree = mTrees.get(0);
         updateHTMLPageVersionTS(mTree.getHtmlPage());
         refreshPageInCache(mTree.getHtmlPage());
 
     }
 
+    /**
+     * Save a collection of {@link MultiTree} and link them with a page, Also delete all the {@link MultiTree} linked
+     * previously with the page.
+     *
+     * @param pageId Page's identifier
+     * @param mTrees
+     * @throws DotDataException
+     */
+    @WrapInTransaction
+    public static void saveMultiTrees(String pageId, List<MultiTree> mTrees) throws DotDataException {
+        if (mTrees == null || mTrees.isEmpty()) {
+            throw new DotDataException("empty list passed in");
+        }
 
+        Logger.debug(MultiTreeFactory.class, String.format("Saving page's content", mTrees));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') String contains no format specifiers. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)

Reply:
it is true, there is not any wildcard